jbpm-workitems/jbpm-workitems-rest/src/main/java/org/jbpm/process/workitem/rest/RESTWorkItemHandler.java

```diff
@@ -1306,17 +1315,28 @@ protected HttpClient getHttpClient(Integer readTimeout,
 
             return cachedClient;
         }
+        
 
+        String cookie = (String) params.get(PARAM_COOKIE);
+        String url = (String) params.get("Url");
+      
         if (HTTP_CLIENT_API_43) {
             RequestConfig config = RequestConfig.custom()
                     .setSocketTimeout(readTimeout)
                     .setConnectTimeout(connectTimeout)
                     .setConnectionRequestTimeout(connectTimeout)
                     .build();
 
-            HttpClientBuilder clientBuilder = HttpClientBuilder.create()
-                    .setDefaultRequestConfig(config);
-            
+            HttpClientBuilder clientBuilder;
+            if (cookie!=null && !cookie.isEmpty()) {
+                clientBuilder = HttpClientBuilder.create()
+                        .setDefaultCookieStore(addCookie(cookie,url))
+                        .setDefaultRequestConfig(config);
+            } else {
+                clientBuilder = HttpClientBuilder.create()
+                        .setDefaultRequestConfig(config);
+            }
```

Comment:
It could be written without "else" clause:
```suggestion
            clientBuilder = HttpClientBuilder.create()
                       .setDefaultRequestConfig(config);
            if (cookie!=null && !cookie.isEmpty()) {
                clientBuilder.setDefaultCookieStore(addCookie(cookie,url))
            } 
```